nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ValidateJson.java

```diff
@@ -0,0 +1,159 @@
+public class ValidateJson extends AbstractProcessor {
+
+    public static final PropertyDescriptor SCHEMA_FILE = new PropertyDescriptor.Builder()
+            .name("Schema File")
+            .description("The path to the Schema file that is to be used for validation")
+            .required(true)
+            .addValidator(StandardValidators.FILE_EXISTS_VALIDATOR)
+            .build();
+
+    public static final Relationship REL_VALID = new Relationship.Builder()
+            .name("valid")
+            .description("FlowFiles that are successfully validated against the schema are routed to this relationship")
+            .build();
+    public static final Relationship REL_INVALID = new Relationship.Builder()
+            .name("invalid")
+            .description("FlowFiles that are not valid according to the specified schema are routed to this relationship")
+            .build();
+
+    private List<PropertyDescriptor> properties;
+    private Set<Relationship> relationships;
+    private final AtomicReference<Schema> schemaRef = new AtomicReference<>();
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> properties = new ArrayList<>();
+        properties.add(SCHEMA_FILE);
+        this.properties = Collections.unmodifiableList(properties);
+
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(REL_VALID);
+        relationships.add(REL_INVALID);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @OnScheduled
+    public void parseSchema(final ProcessContext context) throws IOException {
+        try(FileInputStream inputStream = new FileInputStream(new File(context.getProperty(SCHEMA_FILE).getValue()))) {
+            JSONTokener jsonTokener = new JSONTokener(inputStream);
+            JSONObject jsonObject = new JSONObject(jsonTokener);
+            Schema schema = SchemaLoader.load(jsonObject);
+            this.schemaRef.set(schema);
+        }
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) {
+        final List<FlowFile> flowFiles = session.get(50);
+        if (flowFiles.isEmpty()) {
+            return;
+        }
+        final Schema schema = schemaRef.get();
+        final ComponentLog logger = getLogger();
+
+        for (final FlowFile flowFile : flowFiles) {
+            final AtomicBoolean valid = new AtomicBoolean(true);
+            session.read(flowFile, new InputStreamCallback() {
+                @Override
+                public void process(final InputStream in) throws IOException {
```

Comment:
An IOException here looks like it would cause a session rollback, is that your intended behavior? Or should something catch that and route to invalid or failure?